react/features/rtcstats/RTCStats.ts

```diff
@@ -103,29 +105,51 @@ class RTCStats {
      */
     init(options: InitOptions) {
 
-        const { endpoint, meetingFqn, useLegacy, pollInterval, sendSdp } = options;
-
-        const traceOptions = {
-            endpoint,
-            meetingFqn,
-            onCloseCallback: this.handleTraceWSClose.bind(this),
-            useLegacy
-        };
-
-        this.trace = traceInit(traceOptions);
-        if (!this.isPeerConnectionWrapped) {
-            const rtcstatsOptions = {
-                connectionFilter,
-                pollInterval,
-                useLegacy,
-                sendSdp,
-                eventCallback: this.handleRtcstatsEvent.bind(this)
+        const { useLibRtcStats, meetingFqn } = options;
+
+        if (useLibRtcStats) {
+            logger.info('setting up libRtcStats');
+
+            JitsiMeetJS.rtcstats.events.on(
```

Comment:
THe `rtcstats` object itself should extend from EventEmitter.

Reply:
I think you are from an esthetic point of view right.

But the only way to accomplish this appears to be to have the new RtcStats class extend the EventEmitter class. Which then means RtcStats needs to be manually instantiated somewhere (I'm not sure where is the best point to do that). I failed yesterday to accomplish this, so that the Jitsi-meet code can actually hook up the `on` even listeners.

Long story short: I don't have the time left to accomplish this.